src/mozanalysis/frequentist_stats/sample_size.py

```diff
@@ -12,13 +23,374 @@
+        df["effect_size_base_period"] = df["effect_size_period"] - 7
+        df["rel_effect_size"] = df["effect_size_value"] / df["mean_value"]
+
+        return df[
+            [
+                "relative_effect_size",
+                "effect_size_value",
+                "mean_value",
+                "std_dev_value",
+                "sample_size_per_branch",
+                "population_percent_per_branch",
+                "effect_size_period",
+                "mean_period",
+                "std_dev_period",
+            ]
+        ]
+
+    def get_styled_dataframe(self) -> Styler:
+        """returns styled dataframe for results from
+        empirical_effect_size_sample_size_calc
+
+        Arguments:
+            style (bool): If true, return a cleaned up and formatted pandas Styler.
+            Otherwise return a dataframe
+        Returns:
+            Styler: styled dataframe for visualization
+        """
+        df = self.get_dataframe()
+        return self.style_empirical_sizing_result(df)
+
+
+class SampleSizeCurveResultHolder(ResultsHolder):
+    def __init__(self, *args, **kwargs):
+        """
+        Args:
+            metrics (dict, optional): _description_. Defaults to None.
+            params (dict, optional): _description_. Defaults to None.
+        """
+        super().__init__(*args, **kwargs)
+        # need to modify results so it matches the old format
+        self.raw_df = pd.concat(
+            [el.set_index("effect_size", append=True) for el in self.data.values()]
+        )
+        metrics = [el.name for el in self._metrics]
+        results_dict = {}
+        for el in metrics:
+            results_dict[el] = self.raw_df.loc[el, :].reset_index()
+        self.data = results_dict
+
+    def set_raw_data_stats(self, input_data: pd.DataFrame) -> pd.DataFrame:
+        outlier_percentile = self._params["outlier_percentile"] / 100
+        overall_stats = (
+            input_data[[m.name for m in self._metrics]]
+            .agg(
+                [
+                    "mean",
+                    "std",
+                    lambda d: d[d <= d.quantile(outlier_percentile)].mean(),
+                    lambda d: d[d <= d.quantile(outlier_percentile)].std(),
+                ]
+            )
+            .transpose()
+        )
+        overall_stats.columns = ["mean", "std", "mean_trimmed", "std_trimmed"]
+        overall_stats["trim_change_mean"] = (
+            overall_stats["mean_trimmed"] - overall_stats["mean"]
+        ).abs() / overall_stats["mean"]
+        overall_stats["trim_change_std"] = (
+            overall_stats["std_trimmed"] - overall_stats["std"]
+        ).abs() / overall_stats["std"]
+        self._raw_data_stats = overall_stats
+
+    def get_dataframe(
+        self,
+        input_data: pd.DataFrame = None,
+        show_population_pct: bool = True,
+        simulated_values: list = None,
+        append_stats: bool = False,
+    ) -> pd.DataFrame:
```

Comment:
In keeping with `get_dataframe` for the other result types, I wonder if this should have no args and just return the main results returned by the sizing function. The args to control display can instead by used in `get_styled_dataframe`. Cf comments below

Reply:
I think that'd be a good change.  Maybe we should go one step further and make a `dataframe` property.  I'll work on it